vars/buildYumRepo.groovy

```diff
@@ -0,0 +1,27 @@
+void call(Map args = [:]) {
+    lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+    def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: args.buildManifest))
+    echo "Creating repo file and data for ${args.buildManifest}"
+
+    def filename = buildManifest.build.getFilename()
+    def name = buildManifest.build.name
+    def version = buildManifest.build.version
+    def repoFilePath = "rpm/dist/${filename}"
+
+    sh([
+        'createrepo',
+        "\"${repoFilePath}\"",
+    ].join(' '))
+
+    def repoFileContent = [
+        "[${filename}-staging-${version}-${BUILD_NUMBER}]",
+        "name=${name} ${version} ${BUILD_NUMBER} Staging",
+        "baseurl=${args.baseUrl}/rpm/dist/${filename}/",
+        "enabled=1",
+        "autorefresh=1",
+        "type=rpm-md"
+    ].join('\n')
+
+    writeFile file: "${repoFilePath}/${filename}-artifacts.repo", text: repoFileContent
```

Comment:
The file name better match the content:
```
         ${filename}-staging-${version}.repo
```

Reply:
updated

Reply:
Actually it is better to be `${filename}-${version}.staging.repo`
The production can be `${filename}-${version}.repo`

Reply:
sure